netman/adapters/switches/juniper/base.py

```diff
@@ -294,17 +295,17 @@ def set_interface_native_vlan(self, interface_id, vlan):
         if actual_port_mode is ACCESS:
             raise InterfaceInWrongPortMode("access")
         elif actual_port_mode is None:
-            update_attributes.append(self.custom_strategies.set_interface_port_mode_update_element("trunk"))
+            trunk_mode = self.custom_strategies.set_interface_port_mode_update_element("trunk")
 
         if vlan in interface.trunk_vlans:
             raise VlanAlreadyInTrunk(vlan)
         elif interface.trunk_native_vlan != vlan:
-            update_attributes.append(to_ele("<native-vlan-id>{}</native-vlan-id>".format(vlan)))
+            native_vlan_id_node = to_ele("<native-vlan-id>{}</native-vlan-id>".format(vlan))
 
-        if update_attributes:
+        if native_vlan_id_node is not None:
 
             update = Update()
-            update.add_interface(interface_update(interface_id, "0", update_attributes))
+            update.add_interface(self.custom_strategies.interface_native_vlan_id_update(interface_id, "0", native_vlan_id_node, trunk_mode))
```

Comment:
what id the custon stretagy was shorter.

somethign like :
interface = some_empty_update_interface(port_mode_node)
self.custom_strategies.add_native_vlan_id(interface_id)
update = Update()
update.add_interfacE(interface)

this way the job of the custom strategy is small and simple : add the native vlan id node somewhere in an update node